mediatek: CMCC RAX3000M: add support for the new hw combos

csharper2005 opened the pull request:
Support for new spi-nand, RAM and switch combos:
- uboot-mediatek: CMCC RAX3000: add ddr3 build
- uboot-mediatek: CMCC RAX3000: add Airoha AN8855 switch support
- arm-trusted-firmware-mediatek: add FudanMicro manufacturer
- uboot-mediatek: add support for FudanMicro FM25S01A
- kernel: mediatek: add support for FudanMicro FM25S01A
- mediatek: CMCC RAX3000M: add RAX3000Me as alt model

1715173329 committed:
kernel: add support for FudanMicro FM25S01A SPI-NAND

Add support for FudanMicro FM25S01A SPI NAND.
Datasheet: http://eng.fmsh.com/nvm/FM25S01A_ds_eng.pdf

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>
[add lore.kernel.org link to the patch files]
Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19760
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

csharper2005 committed:
uboot-mediatek: CMCC RAX3000M: add Airoha AN8855 switch support

This commit adds support for Airoha AN8855 switch. Some CMCC RAX3000Me hw
revisions are shipped with Airoha switch.

Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19760
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

csharper2005 committed:
uboot-mediatek: CMCC RAX3000M: add ddr3 build

This commit adds ddr3 build for the ddr3 variant of the CMCC RAX3000Me
router.

Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19760
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

ptpt52 commented:
mark

csharper2005 committed:
arm-trusted-firmware-mediatek: add FudanMicro manufacturer

Add FudanMicro manufacturer.

Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19760
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

csharper2005 committed:
uboot-mediatek: add support for FudanMicro FM25S01A

This patch adds support for FudanMicro FM25S01A SPI NAND. It's required
for some CMCC RAX3000Me hardware revisions.

The patch was partially taken from ImmortalWrt.
Link:
https://raw.githubusercontent.com/immortalwrt/immortalwrt/refs/heads/master/package/boot/uboot-mediatek/patches/342-mtd-spinand-Support-fmsh.patch

Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19760
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
This PR and https://github.com/openwrt/openwrt/pull/19777 are pretty similar. 

How do we want to continue with this?

csharper2005 commented:
> This PR and #19777 are pretty similar.
> How do we want to continue with this?

I like both options, they get the job done. Personally, I prefer the more concise one.
              | Changes | Advantages
--           | --            | --
This  PR | - add 2 new u-boots to the existing 2 devices | Less load of the project   infrastructure, less filogic target build time
#19777  | - add 2 new devices & 4 new u-boots | If I understood @1715173329 correctly: Less likely that a RAX3000M (not Me) user make a mistake and flash ddr3 u-boot instead of ddr4 one.

Users should read manuals. They may as well make a mistake right now, before the changes. For example, flash `emmc` u-boot on the `nand` device.

csharper2005 committed:
mediatek: CMCC RAX3000M: add RAX3000Me as alt model

The devices are basically identical. The RAX3000Me can be with
ddr3 RAM.

Signed-off-by: Mikhail Zhilkin <csharper2005@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19760
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

csharper2005 commented:
Changes:
- Rebase & refresh patches

hauke commented on target/linux/generic/pending-6.12/489-mtd-spinand-add-FudanMicro-FM25S01A.patch:
This patch was merged upstream, please use that version and place it into the backports directory.

csharper2005 replied:
Ok. I'll replace the commit by the @1715173329's one.

hauke commented on package/boot/uboot-mediatek/patches/437-add-cmcc_rax3000m.patch:
```diff
@@ -321,7 +321,7 @@
 +	mediatek,gmac-id = <0>;
 +	phy-mode = "2500base-x";
-+	mediatek,switch = "mt7531";
++	mediatek,switch = "auto";
```
The other PR does not change this. @1715173329  did you miss this?

csharper2005 commented:
> Is this devices used by consumers or is it a board used more by experts?

This is a router for the China Mobile consumers, not a special board. But in some cases (the instruction with config decryption does not consistently succeed) only an expert can install OpenWrt.

<img width="393" height="345" alt="image" src="https://github.com/user-attachments/assets/184ef97a-89ac-4f8d-a2b7-4d5f22e8b4de" />
